src/validation.h

```diff
@@ -600,7 +601,10 @@ class CChainState
     //! CChainState instances.
     BlockManager& m_blockman;
 
-    explicit CChainState(CTxMemPool& mempool, BlockManager& blockman, std::optional<uint256> from_snapshot_blockhash = std::nullopt);
+    explicit CChainState(
+        CTxMemPool* const mempool,
```

Comment:
These new `const` keywords on the `CChainstate::CChainstate` and `InitializeChainstate` arguments are harmless but unnecessary. When you're passing by value (which is what happens when you pass a pointer), making that argument const doesn't change the interface, since the argument is copied to a locally scoped parameter inside the function. It's only when passing by reference that marking arguments const matters.

You can tell that it doesn't make a difference because the function declaration here has the first arg as `CTxMemPool* const`, and the function definition in the .cpp file has `CTxMemPool*` (non const pointer). The compiler doesn't complain because the const-ness of a pass by value arg isn't part of the signature.

Reply:
> this makes it clear that we aren't going to do any pointer arithmetic on mempool

`mempool` inside the function is a copy of the pointer, and so whether or not it's mutated is not part of the contract with the caller, and so doesn't need to be in the function declaration. Like I said it's harmless, but you'll see that there are almost no examples in the code of pass by value parameters being marked const.